src/microbe_stage/editor/MicrobeEditorGUI.cs

```diff
@@ -1346,6 +1321,32 @@ private void UpdateSymmetryIcon()
         }
     }
 
+    /// <summary>
+    ///   Lock / unlock a single organelle that need a nucleus
+    /// </summary>
+    private void UpdatePartAvailability(bool hasNucleus, MicrobePartSelection item)
+    {
+        // item could be a non-implemented organelle so check if it really exists first.
+        if (!SimulationParameters.Instance.DoesOrganelleExist(item.Name))
+        {
+            item.Locked = true;
+            return;
+        }
+
+        if (item.Name == nucleus.InternalName)
+        {
+            item.Locked = hasNucleus;
+        }
+        else if (SimulationParameters.Instance.GetOrganelleType(item.Name).RequiresNucleus)
```

Comment:
Why can't the organelle definition be saved in the item? For example when it is created. Then the above check on line 1330 could be just a simple null check to detect unimplemented ones.

Reply:
The part selection class is not specific to organelles, it's also used by membrane types so for more general use it only rely on names for reference. Maybe instead have the part selection lists turned into dictionaries where the key as the part type?

Reply:
That would also work

Reply:
Actually I think part type as the value is might the right way around.

Reply:
Using the instances as the keys is a bit weird to me. It's already done elsewhere that a registry type of object is used as a key. I think there is easily enough accessible values if you need them specifically from a dictionary.

Reply:
Alright, switched it.